Expensify/App
# Close attachment modal on escape

### Details
~~- Add `componentDidMount` and `componentWillUnmount` lifecycle methods to `AttachmentModal`~~
    ~~- `componentDidMount` registers a keypress event listener~~
        ~~- If the escape key (Key Code 27) is pressed and the attachment modal is open, close the attachment modal.~~
    ~~- `componentWillUnmount` removes the event listener~~
- Add lifecycle methods to `Modal` with `useEffect` 
    - mounting registers a keypress event listener
        - If the escape key (Key Code 27) is pressed, close the attachment modal.
    - unmounting removes the event listener
    

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/1039

### Tests
**Test 1**
1. Click the paperclip icon
2. Select a file to upload
3. Before confirming the upload, press Escape
4. Observe that the upload modal stays on the screen

**Test 2**
1. Click the paperclip icon
2. Select an image to upload
3. Upload the file
4. Click into the image
5. Press escape
6. Observe that the attachment modal stays on the screen

### Tested On


### Screenshots

#### Web
no changes

#### Mobile Web
no changes

#### Desktop
no changes

#### iOS
no changes

#### Android
no changes
